src/libstd/sys/unix/thread.rs

```diff
@@ -281,27 +255,52 @@ pub unsafe fn set_name(name: &str) {
           target_os = "bitrig",
           target_os = "openbsd"))]
 pub unsafe fn set_name(name: &str) {
-    // pthread_set_name_np() since almost forever on all BSDs
+    extern {
+        fn pthread_set_name_np(tid: libc::pthread_t, name: *const libc::c_char);
+    }
     let cname = CString::new(name).unwrap();
     pthread_set_name_np(pthread_self(), cname.as_ptr());
 }
 
 #[cfg(any(target_os = "macos", target_os = "ios"))]
 pub unsafe fn set_name(name: &str) {
-    // pthread_setname_np() since OS X 10.6 and iOS 3.2
+    extern {
+        fn pthread_setname_np(name: *const libc::c_char) -> libc::c_int;
+    }
     let cname = CString::new(name).unwrap();
     pthread_setname_np(cname.as_ptr());
 }
 
 pub unsafe fn join(native: rust_thread) {
-    assert_eq!(pthread_join(native, ptr::null_mut()), 0);
+    eq!(pthread_join(native, ptr::null_mut()), 0);
 }
 
 pub unsafe fn detach(native: rust_thread) {
-    assert_eq!(pthread_detach(native), 0);
+    eq!(pthread_detach(native), 0);
 }
 
-pub unsafe fn yield_now() { assert_eq!(sched_yield(), 0); }
+pub unsafe fn yield_now() {
+    eq!(sched_yield(), 0);
+}
+
+pub fn sleep(dur: Duration) {
+    unsafe {
+        if dur < Duration::zero() {
```

Comment:
@alexcrichton @nagisa @sfackler 

So, my basic thought there is echoing @alexcrichton's, namely that in the long run it's not clear whether we'll even allow negative durations (that's been a complaint for a while).

If we do keep negative durations around, I'm a bit on the fence for whether to panic or yield; I'm sympathetic to the arguments on both sides. I would probably lean toward yielding, because:

* It's a reasonable interpretation of the input,
* It's a smaller hammer than panicking (which we do very rarely in `std` now), and
* It's not clear to me that there is really a footgun here we need to protect against.

In particular, I buy @sfackler's argument that it's easy to wind up with a sensible but negative duration and pass it along, and in most cases the yielding interpretation makes sense there.

@nagisa, OTOH, if you can point out some compelling situations where it's easy to make a bug here and yielding instead of sleeping would be a big problem, that could be helpful.